apps/prairielearn/elements/pl-code/pl-code.py

```diff
@@ -141,8 +142,9 @@ def get_formatter(
     BaseStyle: type[pygments.style.Style], highlight_lines_color: str | None
 ) -> HighlightingHtmlFormatter:
     class CustomStyleWithAnsiColors(BaseStyle):
-        styles = dict(BaseStyle.styles)
-        styles.update(get_ansi_color_tokens())
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+            self.styles = {**self.styles, **get_ansi_color_tokens()}
```

Comment:
This change broke the display of ANSI colors in text. Here's the `element/code` question in `exampleCourse` on this branch (no colors):

<img width="950" alt="Screenshot 2025-01-07 at 14 13 34" src="https://github.com/user-attachments/assets/ca050b56-62f7-45c1-8aa5-31c0977c659a" />

And on master (colors):

<img width="950" alt="Screenshot 2025-01-07 at 14 15 05" src="https://github.com/user-attachments/assets/98a53c9c-f9e7-430a-9403-6a6012076b6a" />

Reply:
Ok, let me look into it further.

Reply:
Ok so pygments is at fault here: they didn't annotate their mutable variable, so pyright thinks it's an instance variable.

Reply:
Then, when we annotate our mutable variable, pyright thinks we are overriding their instance variable with our class variable.

Reply:
So changed back to the `type: ignore` and added a comment explaining why in [04cb7db](https://github.com/PrairieLearn/PrairieLearn/pull/11111/commits/04cb7db09ac09e1dc001aab3dac904e2987fba22).

Reply:
Thank you for testing my change to `__init__`... for me @nwalters512 .

Reply:
Thanks for looking into that. A few notes:

- Can we instead disable the Ruff rule for that line? I _really_ don't like disabling the type checker if we can avoid it.
- Did you consider opening a PR to fix this upstream? I try to do that whenever I encounter basic types issus with packages we use.

Reply:
- Resolved in a335808.
- https://github.com/pygments/pygments/pull/2838

Reply:
Thanks for opening that. That's a biiiiig diff, you'll probably have a much easier time getting a fix merged if e.g. you enabled just https://docs.astral.sh/ruff/rules/mutable-class-default/ (`RUF012`) and included fixes for only that.

Reply:
Those changes are only from RUF012 🤣